### PR TITLE
Use HTTPS links for URLs in READMEs / comments / docs

### PR DESCRIPTION
**Pull request by ewindisch**
Changes all HTTP links to HTTPS links where appropriate
in comments, README files, LICENSE files, man pages, 
and top-level documentation. Does not change the bulk of
documentation which will follow in a separate PR, nor does
it change any use of HTTP URLs in the source itself besides
where such URLs may be simply printed as usage.

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

**Comment by ehazlett** [comment]
LGTM

File: daemon/daemon.go [hunk 1]
```diff
@@ -957,7 +957,7 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine, registryService
 	localCopy := path.Join(config.Root, "init", fmt.Sprintf("dockerinit-%s", dockerversion.VERSION))
 	sysInitPath := utils.DockerInitPath(localCopy)
 	if sysInitPath == "" {
-		return nil, fmt.Errorf("Could not locate dockerinit: This usually means docker was built incorrectly. See http://docs.docker.com/contributing/devenvironment for official build instructions.")
+		return nil, fmt.Errorf("Could not locate dockerinit: This usually means docker was built incorrectly. See https://docs.docker.com/contributing/devenvironment for official build instructions.")
```
Comment (thaJeztah) [inline daemon/daemon.go]: Just noticed this link is outdated (it *does* redirect, so no biggie). New URL that it redirects to is http://docs.docker.com/project/set-up-prereqs/

Comment (thaJeztah) [inline daemon/daemon.go]: Oh, it actually looks like it redirects from `https://....` to `http://..`. Will see if I can find what's causing that and create an issue.

Comment (ehazlett) [inline daemon/daemon.go]: Thanks!

**Pull request merged**